src/core_plugins/elasticsearch/lib/kibana_index_patches/patch_missing_title_keyword_fields.js

```diff
@@ -0,0 +1,68 @@
+import { getRootProperties } from '../../../../server/mappings';
+import { get } from 'lodash';
+import { KibanaIndexPatch } from './kibana_index_patch';
+
+const propertiesWithTitles = [
+  'index-pattern',
+  'dashboard',
+  'visualization',
+  'search',
+];
+
+export class PatchMissingTitleKeywordFields extends KibanaIndexPatch {
+  getUpdatedPatchMappings() {
+    const properties = getRootProperties(this.currentMappingsDsl);
+    const mappings = {};
+
+    for (const property of propertiesWithTitles) {
+      const hasKeyword = !!get(properties, `${property}.properties.title.fields.keyword`);
```

Comment:
I suggest using `getProperty(this.currentMappingsDsl, [property, 'title', 'keyword'])`